plugins/chat/app/services/chat/search_chatable.rb

```diff
@@ -112,26 +86,55 @@ def search_users(context, guardian, contract)
       if context.excluded_memberships_channel_id
         user_search =
           user_search.where(
-            "NOT EXISTS (
-      SELECT 1
-      FROM user_chat_channel_memberships
-      WHERE user_chat_channel_memberships.user_id = users.id AND user_chat_channel_memberships.chat_channel_id = ?
-    )",
+            "NOT EXISTS (SELECT 1 FROM user_chat_channel_memberships WHERE user_id = users.id AND chat_channel_id = ?)",
             context.excluded_memberships_channel_id,
           )
       end
 
       user_search
     end
 
-    def search_groups(context, guardian, contract)
+    def search_groups(context, guardian)
       Group
         .visible_groups(guardian.user)
         .includes(users: :user_option)
         .where(
           "groups.name ILIKE :term_like OR groups.full_name ILIKE :term_like",
           term_like: "%#{context.term}%",
         )
+        .limit(SEARCH_RESULT_LIMIT)
+    end
+
+    def search_category_channels(context, guardian)
+      ::Chat::ChannelFetcher.secured_public_channel_search(
+        guardian,
+        status: :open,
+        filter: context.term,
+        filter_on_category_name: false,
+        match_filter_on_starts_with: false,
+        limit: SEARCH_RESULT_LIMIT,
+      )
+    end
+
+    def search_direct_message_channels(context, guardian, contract, users)
+      channels =
+        ::Chat::ChannelFetcher.secured_direct_message_channels_search(
+          guardian.user.id,
+          guardian,
+          filter: context.term,
+          match_filter_on_starts_with: false,
+          limit: SEARCH_RESULT_LIMIT,
+        ) || []
+
+      # skip 1:1s when search returns users
+      if contract.include_users && users.present?
+        channels.reject! do |channel|
+          other_user_ids = channel.allowed_user_ids - [guardian.user.id]
+          other_user_ids.size <= 1
```

Comment:
This is the _gist_ of this change. When some users are returned as part of the search, we'll automatically skip all 1:1 DMs.

Reply:
Alright, cool let's try it. Do you think we should have a test for this, as a way to describe the expected behaviour?

Reply:
It's [already tested](https://github.com/discourse/discourse/blob/13938984c295148deb720ad94c3af76a24be5ad5/plugins/chat/spec/services/chat/search_chatable_spec.rb#L219-L226). The code that was there before had another `or` condition that wasn't tested and was triggering the issue. I basically removed that condition and refactored the code a bit.